Packaging and compression of files in a distribution

makxdekkers opened the pull request:
I added text for the packaging/compression of files in the distribution. I simplified the proposal by only adding one property: `dcat:wrapFormat`.

I could have created four properties `dcat:compressionFormat`, `dcat:compressionMediaType`, `dcat:packagingFormat` and `dcat:packagingMediaType`, but I thought that would make the solution much too complex. I would think that it would be fairly trivial for the system to decompress or unpack using a single property, and the duplication between mediaType and format seemed unnecessary.

makxdekkers committed:
Update index.html

makxdekkers committed:
Update index.html

agbeltran commented:
Preview here: https://rawgit.com/w3c/dxwg/makxdekkers-patch-1/dcat/index.html

smrgeoinfo commented:
Why is the range dct:MediaTypeOrExtent instead of dct:MediaType?

jakubklimek commented:
I still think that 2 or 4 properties would be cleaner than just the one.

1. The separation between mediaType and format is due to the incompleteness of IANA Media Types list (e.g. does not contain Media Type for `.tar` files) and, therefore, usage of, e.g. File Type EU MDR NAL in DCAT-AP in addition to it. Since we have both properties describing the main content of the distribution, why should the packaging and compression be described by only one of those?
2. Could we go back to #259 and try to represent the various use cases (compressed file, packaged files, compressed and packaged files) using just the wrapFormat property and see how the files can be automatically processed based on that description? I think this will only work for simply compressed file, e.g. `.csv.gz`. We will be unable to determine that the file is actually an archive of multiple files, e.g. `.zip` containing multiple `.csv` files, and how to unpack it (e.g. compared to `.tar.gz` containing multiple `.csv` files). That is why I would rather have a robust solution than a less complex one. I would say that the three use cases are pretty common and it would be nice to be able to properly describe them.

makxdekkers commented:
I am happy to consider the more complex solution with multiple properties, if that is what the group wants. In response to @jakubklimek I would say that 'cleaner' is not the main objective here -- I would say that a more complex solution should be chosen if it is essential and a simpler solution does not work.
Furthermore, I think that even the approach with four properties can only distinguish simple cases; you can only indicate the outermost and the innermost formats. For example, there is no way to know whether a RAR archive contains multiple ZIP files that contain CSV and XLS files.

makxdekkers commented:
@smrgeoinfo I made the range `dct:MediaTypeOrExtent` instead of `dct:MediaType` to align with the range of `dcat:mediaType` and `dct:format`, but you are right, it could easily be `dct:MediaType`.

smrgeoinfo commented:
I was assuming that for a packaged distribution, one could use dct:conformsTo to specify the packaging convention (e.g. bagit, OASIS OpenDocument Package Format...). In those schemes there is a manifest and metadata document in the package that details the internal structure and file formats.

jakubklimek commented:
@smrgeoinfo That may apply to distributions complying with the package formats you mention. However, we could have a more loose package. If we had for instance a tar or zip file with 100 CSV files, one per day, all with the same schema, this schema (CSV on the Web JSON-LD descriptor) would go to `dct:conformsTo` and we need a separate property (or properties) to describe the package/compression.

jakubklimek commented:
@makxdekkers You are right about the simple cases. However, I would say that double compression and similar techniques are often enough just bad practice. I often see compressed single files and packaged (and compressed) sets of files with the same schema where it makes sense (splitting larger files into smaller ones or just reducing size of file on disk). However, when I come across a zip file containing many other zip files, it is often because the publisher does not know what they are doing. The question seems to be where do we draw the line.

makxdekkers commented:
Can I ask the members of the WG to vote for one of the two alternatives:

1. use a single property, `dcat:wrapFormat`, for both compression and packaging, as in the pull request
2. use two properties, e.g. `dcat:compressFormat` and `dcat:packFormat`, to distinguish between the two

Thanks.

larsgsvensson commented:
@makxdekkers How does the vote take place? My suggestion would be one comment for each of the proposals (single property vs two properties) and then use thumbs-up for the vote.

makxdekkers commented:
Proposal 1:  use a single property, `dcat:wrapFormat`, for both compression and packaging, as in the pull request 
Please give thumbs-up if you agree.

makxdekkers commented:
Proposal 2: use two properties, e.g. `dcat:compressFormat` and `dcat:packFormat`, to distinguish between the two
Please give thumbs-up if you agree.

dr-shorthair reviewed (changes requested):
1. These new properties also need to be expressed in the RDF file (i.e. dcat.ttl) 
2. Also add an entry in the change-log (Annex E)

makxdekkers committed:
Update index.html

Commit:
Merge branch 'gh-pages' into makxdekkers-patch-1

Commit:
Merge branch 'gh-pages' into dcat-makx-packaging

Commit:
Merge branch 'dcat-makx-packaging' into makxdekkers-patch-1

# Conflicts:
#	dcat/index.html

Commit:
remove stray merge remnant

makxdekkers committed:
Update index.html

makxdekkers committed:
New properties for compression and packaging

Added the compressFormat and packageFormat properties in the text and in the TTL

agbeltran commented:
Linking here the related issues: #54, #259, #482

We should also add examples to cover the different use cases.

agbeltran committed:
Adding dcat:compressFormat to dcat.ttl

agbeltran committed:
Adding dcat:packageFormat to dcat.ttl

agbeltran committed:
Adding change log

agbeltran committed:
Merging second branch: dcat-makx-packaging

agbeltran commented on dcat/index.html:
```diff
@@ -62,12 +62,12 @@
 
     <h3 id="dcat_history">DCAT history</h3>
     <p>The original DCAT vocabulary was developed and <a href="http://vocab.deri.ie/dcat">hosted</a> at the Digital Enterprise Research Institute (DERI), then refined by the <a href="http://www.w3.org/egov/">eGov Interest Group</a>, and finally standardized in 2014 [[?VOCAB-DCAT-20140116]] by the <a href="http://www.w3.org/2011/gld/">Government Linked Data (GLD)</a> Working Group.</p>
-    <p>This revised version of DCAT was developed by the <a href="https://www.w3.org/2017/dxwg/">Dataset Exchange Working Group</a> in response to a new set of Use Cases and Requirements [[?DCAT-UCR]] submitted on the basis of experience with the DCAT vocabulary from the time of the original version, and new applications not originally considered. A summary of the changes from  [[?VOCAB-DCAT-2014 ...
```
This needs to be rolled back

agbeltran replied:
done in 67d9e3b

agbeltran commented on dcat/index.html:
```diff
@@ -4073,6 +4107,9 @@ <h3>Changes since the W3C Recommendation of 16 January 2014</h3>
                 Each relevant class section was extended with properties to deal with licensing, acccess rights, other rights and policies associated with the classes, as
                 recommended in the  <a href="#license-rights">License and rights statements</a> section.
             </li>
+            <li>
```
This needs to be removed: it is repeated because I merged the two branches that had similar changes

agbeltran replied:
done in 5b88e8a

agbeltran committed:
Removing duplicate addition in change log

agbeltran committed:
Rolling back incorrect change of 'Change History' for 'compression'

agbeltran commented:
@makxdekkers I merged [the other branch](https://github.com/w3c/dxwg/tree/dcat-makx-packaging) into this PR

@dr-shorthair if you could have a final check and merge, we resolved this on the call today: https://www.w3.org/2019/02/20-dxwgdcat-minutes.html#x13

agbeltran commented:
the remaining work on this is to add the examples

dr-shorthair commented:
Merging for now, noting that instance examples also needed. 
Working on the assumption that the discussion thread above is sufficient endorsement by the team.

dr-shorthair commented:
Hmmm. This merge was probably premature. I don't think we have a decision recorded to support it. My bad. And not competent to unwind ...

agbeltran commented:
We did decide about this on the call yesterday: https://www.w3.org/2019/02/20-dxwgdcat-minutes.html#x13

dr-shorthair commented:
Phew. It was resolved after I left the meeting and before I read the minutes :-) No reverts needed.
